### PR TITLE
Custom table rendering of PackageRevision resources

### PR DESCRIPTION
**Pull request by martinmaly**

### CODE REVIEW

File: porch/apiserver/pkg/registry/porch/table.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+			{Name: "Latest", Type: "boolean"},
+			{Name: "Lifecycle", Type: "string"},
+			{Name: "Repository", Type: "string"},
+		},
+	}
+)
+
+func isLatest(pr *api.PackageRevision) bool {
+	labels := pr.Labels
+	if labels == nil {
```
Comment (justinsb) [inline porch/apiserver/pkg/registry/porch/table.go:117]: Nit: You don't need to check for nil when _reading_ a slice or map; the nil value is treated as the empty slice/map.

(This is also true when appending to an nil slice.  It is sadly not true when writing to an nil map)

File: porch/apiserver/pkg/registry/porch/table.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+		return false
+	}
+	return labels[api.LatestPackageRevisionKey] == api.LatestPackageRevisionValue
+}
+
+func notAcceptableError(ctx context.Context, resource schema.GroupResource) error {
```
Comment (justinsb) [inline porch/apiserver/pkg/registry/porch/table.go:123]: Nit (for future PR): move these into their own file (errors.go)?

**Review by justinsb — APPROVED** [review]

File: porch/apiserver/pkg/registry/porch/table.go [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+	}
+
+	return &table, nil
+}
+
+var (
+	packageRevisionTableConvertor = tableConvertor{
+		resource: porch.Resource("packagerevisions"),
+		cells: func(obj runtime.Object) []interface{} {
+			pr, ok := obj.(*api.PackageRevision)
```
[the remark this answers is not in the document]
Comment (martinmaly) [reply porch/apiserver/pkg/registry/porch/table.go:91]: we don't ... the caller will interpret this as an error and return an error. I wanted the per-resource logic minimal so I hoisted the error handling out.

File: porch/apiserver/pkg/registry/porch/table.go (hunk 1 above)
Comment (martinmaly) [reply porch/apiserver/pkg/registry/porch/table.go:117]: You are right! indeed, the asymmetry of map got me there.

File: porch/apiserver/pkg/registry/porch/table.go (hunk 2 above)
Comment (martinmaly) [reply porch/apiserver/pkg/registry/porch/table.go:123]: did it now. thanks!

**Pull request merged**